### ISSUE TITLE
2019.07 Math Sequences Squashathon

### ISSUE THREAD

**Issue by AlexDaniel** [ajs/perl6-Math-Sequences#30]
Hello! I saw [your blog post](http://ajs.github.io/tools/2019/06/12/perl6-integer-sequences.html) and I think it's a great topic for the [upcoming squashathon](https://github.com/rakudo/rakudo/wiki/Monthly-Bug-Squash-Day). Currently the topic for 2019.07 is set to something else, and we're discussing if it'd make sense to change it. Obviously to make it happen we'll need you to be on board, and I'll likely need admin privileges for this repo so that I can set and maintain the webhook for a bot (during the event).

Basically, the idea is to make a list of sequences that are not implemented yet, and distribute them across contributors. As usual, the most prominent contributor gets a plush camelia.

What do you think?

**Comment by ajs** [comment]
Sounds great! I'm completely on board. However, my work demands are high and I can never be sure when I'll have time to be involved. As long as you don't need me directly, that's fine. As for admin privs, I can certainly add you.

One issue: there are some sequences that are very hard (either in terms of implementation or computation or both). It's important to select carefully. The OEIS entries actually have keywords "hard" and "easy" for this purpose.

**Comment by taboege** [comment]
Count me in if this is going to be the squashathon this month. It's not my area of expertise (at all!), but at least I hope to get some credit as a math PhD student with an impure, practical interest in integer sequences and the OEIS in particular. It might be appropriate to mention my [OEIS module](https://github.com/taboege/p6-OEIS) and that the "easy" and "hard" keywords are implemented there and smartmatch:

``` perl6
OEIS::lookup-all(1, 1, * + * ... *).grep(* !~~ OEIS::easy).head.say;
#= OEIS A290689 «Number of transitive rooted trees with n nodes.»
```
`</plug>`. Might be useful as a CLI tool during the squashathon?

**Comment by JJ** [comment]
The squashathon just started!

**Comment by JJ** [comment]
Has the hook been set already?

**Comment by AlexDaniel** [comment]
> and I'll likely need admin privileges for this repo so that I can set and maintain the webhook for a bot (during the event).

I don't have that. Can't set up the hook right now. To do it, an admin of this repo should:

* Go to https://github.com/perl6/perl6-Math-Sequences/settings/hooks/new
* Set the settings like this:
![sceenshot](https://user-images.githubusercontent.com/5507503/46384999-261a1a00-c6c2-11e8-9c1b-7295736d4004.png)
* Where the secret is `squashathonsequences`

**Comment by AlexDaniel** [comment]
Ping @ajs.

**Comment by AlexDaniel** [comment]
OK, I guess the webhook is not happening, but it's alright. I'll be merging a bunch of PRs so that it's easier to see which sequences are still not implemented. We'll review the results manually later!

**Comment by AlexDaniel** [comment]
I did my best to compile a list of contributions. In total there were 60 sequences (plus some hardcoded ones) that were implemented during this squashathon!

Here is the list:

* @tbrowder
  * \+ ASCII equivalents for integer sequences

* @lancew
  * A001006

* @MasterDuke17 (MasterDuke)
  * A000111
  * A000364
  * A001003
  * A001519
  * A004011
  * A006894

* @taboege (tobs)
  * A000029
  * A000031
  * A000048
  * A001037
  * A001481
  * A008683
  * A025487
  * \+ A lot of useful subroutines

* @AlexDaniel
  * A000594
  * A000793
  * A000959 (but broken)
  * A001190
  * A002083
  * A002654
  * A005230
  * A074206
  * \+ hardcoded versions of `more` sequences (A000109, A000112, A000609, A002106, A003094, A005470, A006966)
  * \+ whitespace and .pm6/.t6 changes
  * \+ a script to populate the issue tracker

* @thundergnat
  * A000010
  * A000110
  * A000161
  * A000166
  * A000169
  * A000182
  * A000262
  * A000311
  * A000312
  * A000326
  * A000330
  * A000396
  * A000593
  * A000688
  * A000961
  * A001045
  * A001113
  * A001285
  * A001333
  * A001699
  * A001906
  * A002110
  * A002113
  * A002572
  * A002658
  * A003136
  * A005117
  * A005130
  * A005811
  * A006318
  * A008277
  * A010060
  * A027641
  * A027642
  * A038566
  * A038567
  * A038568
  * A038569
  * \+ A fix to disambiguify A002572 and A005588
  * \+ A lot of useful subroutines

## :tada::tada: And the winner is @thundergnat! :tada::tada: (Please contact @lizmat to get your plush camelia!)

See you all next month in another repository :)
